File: projeto 2/academia/aspnet-backend/AcademiaAPI/Models/Professor.cs
using System.ComponentModel.DataAnnotations;

namespace AcademiaAPI.Models
{
    public class Professor
    {
        [Key]
        public int Id { get; set; }

        [Required(ErrorMessage = "Nome é obrigatório")]
        [MaxLength(100)]
        public string Nome { get; set; } = string.Empty;

        [Required(ErrorMessage = "CPF é obrigatório")]
        [MaxLength(14)]
        public string Cpf { get; set; } = string.Empty;

        [Required(ErrorMessage = "Email é obrigatório")]
        [EmailAddress]
        [MaxLength(150)]
        public string Email { get; set; } = string.Empty;

        [Required(ErrorMessage = "Especialidade é obrigatória")]
        [MaxLength(100)]
        public string Especialidade { get; set; } = string.Empty;

        [Required(ErrorMessage = "CREF é obrigatório")]
        [MaxLength(30)]
        public string Cref { get; set; } = string.Empty;

        [MaxLength(20)]
        public string? Telefone { get; set; }

        public bool Ativo { get; set; } = true;

        public DateTime CriadoEm { get; set; } = DateTime.UtcNow;


        public ICollection<Matricula> Matriculas { get; set; } = new List<Matricula>();
    }
}

File: projeto 2/academia/laravel-frontend/app/Models/Professor.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class Professor extends Model
{
    use HasFactory;

    protected $fillable = [
        'nome',
        'cpf',
        'email',
        'especialidade',
        'cref',
        'telefone',
        'ativo',
    ];

    protected $casts = [
        'ativo' => 'boolean',
    ];

    public function matriculas()
    {
        return $this->hasMany(Matricula::class);
    }

    public function alunos()
    {
        return $this->hasManyThrough(Aluno::class, Matricula::class, 'professor_id', 'id', 'id', 'aluno_id');
    }
}
